public/model/layerRenderController.ts

```diff
@@ -22,16 +22,25 @@ interface MaplibreRef {
   current: Maplibre | null;
 }
 
-// OpenSearch only accepts longitude in range [-180, 180]
-// Maplibre could return value out of the range
-function adjustLongitudeForSearch(lon: number) {
-  if (lon < -180) {
-    return -180;
-  }
-  if (lon > 180) {
-    return 180;
+// calculate lng limits based on map bounds
+// maps can render more than 1 copies of map at lower zoom level and displays
+// one side from 1 copy and other side from other copy at higher zoom level if
+// screen crosses internation dateline
+function calculateBoundingBoxLngLimit(bounds: LngLatBounds) {
+  const boundsMinLng = bounds.getNorthWest().lng;
+  const boundsMaxLng = bounds.getSouthEast().lng;
+  // if bounds expands more than 360 then, consider complete globe is visible
+  if (boundsMaxLng - boundsMinLng >= MAX_LONGITUDE - MIN_LONGITUDE) {
+    return {
+      right: MAX_LONGITUDE,
+      left: MIN_LONGITUDE,
+    };
   }
-  return lon;
+  // wrap bounds if only portion of globe is visible
+  return {
+    right: bounds.getSouthEast().wrap().lng,
```

Comment:
I have a question here, suppose the current map lng bounding is `[-170, 181]`, it is `[-170, -179]` after wrapping. It will return 
```
{
  right: -179,
  left: -170,
}
```
The query would be
```
{
  bottom_right: {
        lon: -179,
        lat: ...,
      },
      top_left: {
        lon: -170,
        lat: ...,
      },
    };
```
Is this what OpenSearch expecting?

Reply:
Sorry, do you mean will OpenSearch accept similar to following query or not
```
{
  "DestLocation": {
    "bottom_right": {
      "lon": -176.8581559092076,
      "lat": -38.91257159667413
    },
    "top_left": {
      "lon": -173.24641047835382,
      "lat": 78.53889135857452
    }
  }
}
```

Reply:
@VijayanB I mean will OpenSearch return the same results as expected in the bounding box `[-170, 181]` after the `lng` been wrapped in such case where `bottom_right.lon < top_left .lon`?

Reply:
@ruanyl Yes.